### PR TITLE
illumos-userland 6974 - add http-parse library

### PR DESCRIPTION
**Pull request by rtvd**
https://www.illumos.org/issues/6974

### CODE REVIEW

File: components/library/http-parser/Makefile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# Copyright 2016 Denys Rtveliashvili.  All rights reserved.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		http-parser
+COMPONENT_VERSION=	2.1
+COMPONENT_FMRI=		library/http-parser
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_PROJECT_URL=	https://packages.qa.debian.org/h/http-parser.html
```
Comment (pyhalov) [inline components/library/http-parser/Makefile]: Please, link to original URL , for example https://github.com/nodejs/http-parser

File: components/library/http-parser/Makefile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:749c35dc4696be4be9bb4f3b388197151cee6e91b5e8895789be904386e38187
+COMPONENT_ARCHIVE_URL=	http://http.debian.net/debian/pool/main/h/http-parser/http-parser_2.1.orig.tar.gz
```
Comment (pyhalov) [inline components/library/http-parser/Makefile]: There's already 2.7.0 release (https://github.com/nodejs/http-parser/releases), why do you want so old package?

File: components/library/http-parser/http-parser.p5m [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 <contributor>
```
Comment (pyhalov) [inline components/library/http-parser/http-parser.p5m]: please, add your name

File: components/library/http-parser/Makefile (hunk 2 above)
Comment (rtvd) [inline components/library/http-parser/Makefile]: Thank you for spotting that.
I tried 2.7.0 but it does not build properly. The bug is a known one and the relevant code is not obvious. So I have refreshed the build to use 2.6.2 instead.

File: components/library/http-parser/Makefile (hunk 1 above)
Comment (rtvd) [inline components/library/http-parser/Makefile]: Done

File: components/library/http-parser/Makefile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+COMPONENT_NAME=		http-parser
+COMPONENT_VERSION=	2.6.2
+COMPONENT_FMRI=		library/http-parser
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_PROJECT_URL=	https://github.com/nodejs/http-parser
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:80fffc3b64ef6968cecdd4b299a96986007dff4bd12ae6c58cbcb506959b90ad
+COMPONENT_ARCHIVE_URL=	https://github.com/nodejs/http-parser/archive/v2.6.2.tar.gz
```
Comment (xen0l) [inline components/library/http-parser/Makefile]: https://github.com/nodejs/http-parser/archive/v2.6.2.tar.gz -> https://github.com/nodejs/http-parser/archive/v$(COMPONENT_VERSION).tar.gz, so it doesn't have to be changed everytime.

File: components/library/http-parser/Makefile [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+COMPONENT_BUGDB=	$(COMPONENT_FMRI)
```
Comment (xen0l) [inline components/library/http-parser/Makefile]: This is relevant only for userland-gate, not oi-userland and should be removed.

File: components/library/http-parser/Makefile [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+
+COMPONENT_SUMMARY=      A parser for HTTP messages written in C.
```
Comment (xen0l) [inline components/library/http-parser/Makefile]: Why is there a space between COMPONENT_SUMMARY and COMPONENT_LICENSE_FILE? Remove it please.

File: components/library/http-parser/http-parser.p5m [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/http_parser.h
+link path=usr/lib/$(MACH64)/libhttp_parser.so \
+    target=/root/oi-userland/components/library/http-parser/build/prototype/$(MACH)//usr/lib/$(MACH64)/libhttp_parser.so.$(COMPONENT_VERSION)
```
Comment (xen0l) [inline components/library/http-parser/http-parser.p5m]: Wrong target. It should be something target=usr/lib/$(MACH64)/libhttp_parser.so.$(COMPONENT_VERSION)

File: components/library/http-parser/http-parser.p5m [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+file path=usr/lib/$(MACH64)/libhttp_parser.so.$(COMPONENT_VERSION)
+link path=usr/lib/libhttp_parser.so \
+    target=/root/oi-userland/components/library/http-parser/build/prototype/$(MACH)//usr/lib/libhttp_parser.so.$(COMPONENT_VERSION)
```
Comment (xen0l) [inline components/library/http-parser/http-parser.p5m]: Wrong target. It should be something target=usr/lib/libhttp_parser.so.$(COMPONENT_VERSION)

**Comment by xen0l** [comment]
http-parser also has test Makefile target, so you should add it to component makefile as well.

File: components/library/http-parser/Makefile [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+COMPONENT_FMRI=		library/http-parser
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_PROJECT_URL=	https://github.com/nodejs/http-parser
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:80fffc3b64ef6968cecdd4b299a96986007dff4bd12ae6c58cbcb506959b90ad
+COMPONENT_ARCHIVE_URL=	https://github.com/nodejs/http-parser/archive/v$(COMPONENT_VERSION).tar.gz
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+COMPONENT_SUMMARY=      A parser for HTTP messages written in C.
```
Comment (pyhalov) [inline components/library/http-parser/Makefile]: please, remove "." in the end

**Comment by rtvd** [comment]
Does it look good enough for a merge now?

**Comment by pyhalov** [comment]
Yes

**Pull request merged**